sympy/calculus/tests/test_util.py

```diff
@@ -63,6 +62,30 @@ def test_not_empty_in():
         Union(Interval(-2, -1, True, False), Interval(2, oo))
 
 
+def test_periodicity():
+    x = Symbol('x')
+    y = Symbol('y')
+
+    assert periodicity(sin(2*x), x) == pi
+    assert periodicity((-2)*tan(4*x), x) == pi/4
+    assert periodicity(tan(x)*cos(x), x) == 2*pi
+    assert periodicity(sin(x) + cos(x), x) == 2*pi
+    assert periodicity(tan(x) + cot(x), x) == pi
+    assert periodicity(sin(4*x) + sin(x)*cos(x), x) == pi
+    assert periodicity(sin(x) - 1, x) == 2*pi
+    assert periodicity(sin(2*x)*cos(2*x) - y, x) == pi/2
+    raises (NotImplementedError, lambda: periodicity(sin(x) - cos(x), x))
```

Comment:
If it works for `sin(x) + cos(x)` then why not for `sin(x) - cos(x)`. Whats the main problem when coeff is negative?